filebeat/prospector/prospector.go

```diff
@@ -52,16 +56,25 @@ func NewProspector(cfg *common.Config, states file.States, outlet Outlet) (*Pros
 		wg:            sync.WaitGroup{},
 		states:        &file.States{},
 		channelWg:     sync.WaitGroup{},
+		Once:          false,
 	}
 
-	if err := cfg.Unpack(&prospector.config); err != nil {
+	var err error
+	if err = cfg.Unpack(&prospector.config); err != nil {
 		return nil, err
 	}
-	if err := prospector.config.Validate(); err != nil {
+	if err = prospector.config.Validate(); err != nil {
 		return nil, err
 	}
 
-	err := prospector.Init(states)
+	var h map[string]interface{}
+	cfg.Unpack(&h)
+	prospector.Hash, err = hashstructure.Hash(h, nil)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
makes me wonder if we want to provide a `(*common.Config).Hash() ...` method.

Also consider `(*common.Config).ToMap() (map[string]interface{}, error)` and a `(*common.Config).ToArray() ([]interface{}, error)`.

As `*common.Config` can by a dictionary and a list, the hash values from array/dict conversion should be combined (See DebugString in #3304).